### PR TITLE
[R-package] Improves logging in lint_r_code.R (fixes #3471)

### LINKED ISSUE #3471: [R-package] [ci] improve logging in lint_r_code.R
## Summary

We use a script, `.ci/lint_r_code.R`, to check style items in the R package with `{lintr}`. A few improvements could be made to the output from this script.

**what those logs look like right now**

```text
[1] "Found 49 R files to lint\n"
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/.ci/lint_r_code.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/build_r.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/basic_walkthrough.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/boost_from_prediction.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/categorical_features_rules.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/cross_validation.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/early_stopping.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/efficient_many_training.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/leaf_stability.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/multiclass_custom_objective.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/multiclass.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/demo/weight_param.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/inst/make-r-def.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/aliases.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/callback.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.Booster.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.convert_with_rules.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.cv.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.Dataset.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.importance.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.interprete.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.model.dt.tree.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.plot.importance.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.plot.interpretation.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.Predictor.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.train.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lgb.unloader.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/lightgbm.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/metrics.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/readRDS.lgb.Booster.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/saveRDS.lgb.Booster.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/R/utils.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/src/install.libs.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_basic.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_custom_objective.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_dataset.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_learning_to_rank.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_lgb.Booster.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_lgb.convert_with_rules.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_lgb.importance.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_lgb.interprete.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_lgb.plot.importance.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_lgb.plot.interpretation.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_lgb.unloader.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_metrics.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_parameters.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_Predictor.R
[1] Found 0 linting errors in /Users/jlamb/repos/LightGBM/R-package/tests/testthat/test_utils.R
```

## How to fix this issue

1. remove the `\n` in https://github.com/microsoft/LightGBM/blob/7e2f353df4ee6b7a17a653122ab5a549f8e8d0f7/.ci/lint_r_code.R#L89, since it  prints as a literal `\n` and not a newline
    * image
1. Change the first message so it doesn't start with the word "Found", so that it is a little bit more distinguishable from the rest of the log messages
1. use `paste0()` in the first message, instead of `sprintf()`, so the first message's output does not have quotes in it
1. make sure that a confirmation is printed at the end of a successful run, so it is clear that the run succeeded. Something like  adding `print(paste0("Total linting issues: ", issues_found))` after https://github.com/microsoft/LightGBM/blob/7e2f353df4ee6b7a17a653122ab5a549f8e8d0f7/.ci/lint_r_code.R#L114 (up to you what the message is)

## How to test this

Install `{lintr}`

```shell
Rscript -e "install.packages('lintr', repos = 'https://cran.r-project.org')"
```

Run the script

```shell
Rscript .ci/lintr_r_code.R $(pwd)
```

### PR DESCRIPTION
**Pull request by sinAshish**
References Issue  #3471

### CODE REVIEW

File: .ci/lint_r_code.R [hunk 1]
```diff
@@ -86,7 +86,7 @@ LINTERS_TO_USE <- list(
     , "unneeded_concatenation" = lintr::unneeded_concatenation_linter
 )
 
-print(sprintf("Found %i R files to lint\n", length(FILES_TO_LINT)))
+noquote(paste0("LINT INFO: ", length(FILES_TO_LINT), " R files need linting"))
```
Comment (jameslamb) [inline .ci/lint_r_code.R:89]:
```suggestion
noquote(paste0(length(FILES_TO_LINT), " R files need linting"))
```

I don't think the `LINT INFO` is adding any information, so I'd like to suggest removing it

**Review by jameslamb — CHANGES_REQUESTED** [review]
Looks good! I had never heard of `noquote()` before, seems perfect for our use here. Thanks for that!

I just left one small suggestion.

File: .ci/lint_r_code.R (hunk 1 above)
Comment (sinAshish) [reply .ci/lint_r_code.R:89]: ok got it.  I'll make the change

**Review by jameslamb — APPROVED** [review]
Looks great, thank you so much for taking the time to contribute to `LightGBM`! We hope you'll come back and contribute in the future :grinning:

**Comment by sinAshish** [comment]
> Looks great, thank you so much for taking the time to contribute to `LightGBM`! We hope you'll come back and contribute in the future 😀

Thanks. I'm looking for other issues which I can tackle!

**Comment by jameslamb** [comment]
if you'd like another small, focused issue to get more familiar with LightGBM's R package, this is a good one: https://github.com/microsoft/LightGBM/issues/3390

if you want to try a larger issue, #2983 would be very appreciated.

We'd also welcome things that are not explicitly included in issues, like if you see any issues in the documentation at https://lightgbm.readthedocs.io/en/latest/R/reference/

**Pull request merged**